ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/SurfaceMountingManager.java

```diff
@@ -992,9 +993,21 @@ public void updateLayout(int reactTag, int x, int y, int width, int height, int
       parent.requestLayout();
     }
 
-    // TODO: T31905686 Check if the parent of the view has to layout the view, or the child has
-    // to lay itself out. see NativeViewHierarchyManager.updateLayout
-    viewToUpdate.layout(x, y, x + width, y + height);
+    ViewState parentViewState = getViewState(parentTag);
+    ReactViewManagerWrapper parentViewManager = parentViewState.mViewManager;
+    IViewManagerWithChildren parentViewManagerWithChildren;
+    if (parentViewManager instanceof IViewManagerWithChildren) {
+      parentViewManagerWithChildren = (IViewManagerWithChildren) parentViewManager;
+    } else {
+      throw new IllegalViewOperationException(
+          "Trying to use view with tag "
+              + parentTag
+              + " as a parent, but its Manager doesn't implement IViewManagerWithChildren");
+    }
+    if (parentViewManagerWithChildren != null
```

Comment:
Can we simplify this? This null-check would never be hit, as we would've already thrown an exception above?

Alternatively, could we just rely on a simple cast here and leverage the built-in exception?

Reply:
As I mentioned in the description, I copied the code from `NativeViewHierarchyManager` in the old architecture

https://github.com/facebook/react-native/blob/a70354df12ef71aec08583cca4f1fed5fb77d874/ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyManager.java#L210-L221